From 051a1a2ef6e7e3b3ec8571bbbacbdc97cb6c6de1 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 24 Jun 2022 21:59:13 +0200
Subject: [PATCH 26/39] clk: Introduce devm_clk_hw_register_mux_parent_data()

Add an API for clock mux that uses parent_data for the parent instead of
a string parent_name.

Upstream-Status: Submitted [https://patchwork.kernel.org/project/linux-arm-kernel/patch/20220625013235.710346-1-marex@denx.de/]
Signed-off-by: Marek Vasut <marex@denx.de>
Cc: Abel Vesa <abel.vesa@nxp.com>
Cc: Fabio Estevam <festevam@gmail.com>
Cc: Jacky Bai <ping.bai@nxp.com>
Cc: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Cc: Lucas Stach <l.stach@pengutronix.de>
Cc: Michael Turquette <mturquette@baylibre.com>
Cc: Shawn Guo <shawnguo@kernel.org>
Cc: Stephen Boyd <sboyd@kernel.org>
Cc: linux-arm-kernel@lists.infradead.org
Cc: linux-clk@vger.kernel.org
Cc: linux-imx@nxp.com
---
V3: New patch
---
 include/linux/clk-provider.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/linux/clk-provider.h b/include/linux/clk-provider.h
index 267cd06b54a01..1c3561f330259 100644
--- a/include/linux/clk-provider.h
+++ b/include/linux/clk-provider.h
@@ -1047,6 +1047,13 @@ struct clk *clk_register_mux_table(struct device *dev, const char *name,
 	__devm_clk_hw_register_mux((dev), NULL, (name), (num_parents), NULL,  \
 			      NULL, (parent_data), (flags), (reg), (shift),   \
 			      BIT((width)) - 1, (clk_mux_flags), table, (lock))
+#define devm_clk_hw_register_mux_parent_data(dev, name, parent_data,	      \
+					     num_parents, flags, reg, shift,  \
+					     width, clk_mux_flags, lock)      \
+	__devm_clk_hw_register_mux((dev), NULL, (name), (num_parents),	      \
+			      NULL, NULL, (parent_data), (flags), (reg),      \
+			      (shift), BIT((width)) - 1, (clk_mux_flags),     \
+			      NULL, (lock))
 
 int clk_mux_val_to_index(struct clk_hw *hw, const u32 *table, unsigned int flags,
 			 unsigned int val);
-- 
2.39.1

